### PR TITLE
Solution js_advanced_calculator

### PR DESCRIPTION
**Pull request by sergey-lipikhin**

### CODE REVIEW

**Review by timoxa2t — APPROVED** [review]
Good work

File: src/makeCalculator.js [hunk 1]
```diff
@@ -37,7 +37,26 @@
...
+    add: (result, number) => result + number,
+    subtract: (result, number) => result - number,
+    multiply: (result, number) => result * number,
+    divide: (result, number) => result / number,
+
+    operate(callback, number) {
+      this.result = callback(this.result, number);
+
+      return this;
+    },
```
Comment (OleksiiBlezniuk) [inline src/makeCalculator.js:52]:
```suggestion
    add(result, number) {
      return result + number
    },
    
    subtract(result, number) {
      return result - number;
    }
    
    multiply(result, number) {
      return result * number;
    }
    
    divide(result, number) {
      return result / number;
    }

    operate(callback, number) {
      this.result = callback(this.result, number);

      return this;
    },
```

Comment (OleksiiBlezniuk) [reply src/makeCalculator.js:52]: Arrow functions are great, but let's use one approach for methods creating, some of your methods created by arrow functions, some by function declaration

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

**Review by sergii-nosachenko — APPROVED** [review]
Good work 👍

**Pull request closed without merging**